solr/core/src/java/org/apache/solr/servlet/HttpSolrCall.java

```diff
@@ -233,6 +232,8 @@ protected void init() throws Exception {
       path = path.substring(0, alternate.length());
     }
 
+    queryParams = SolrRequestParsers.parseQueryString(req.getQueryString());
```

Comment:
Yea, there is some dance there. But it will fall back to obtaining the correct 400 code and error message from the original exception, so that should be cool.